src/trajectory.jl

```diff
@@ -20,29 +22,31 @@ struct StaticTrajectory{I<:AbstractIntegrator} <: AbstractTrajectory{I}
     n_steps     ::  Int
 end
 
+"""
+Termination (i.e. no-U-turn).
+"""
+struct Termination end
```

Comment:
There is no need to create a new type for numerical errors (since `PhasePoint` can handle it), but we still need a type for other types of terminations, e.g. no-u-turn, maximal tree-depth, and maximal allowed hamiltonian error during leapfrog.